src/main/java/io/appium/java_client/pagefactory/bys/ContentMappedBy.java

```diff
@@ -16,40 +16,45 @@
 
 package io.appium.java_client.pagefactory.bys;
 
-import static io.appium.java_client.pagefactory.utils.WebDriverUnpackUtility.getCurrentContentType;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static io.appium.java_client.pagefactory.bys.ContentType.NATIVE_MOBILE_SPECIFIC;
 
 import org.openqa.selenium.By;
 import org.openqa.selenium.SearchContext;
 import org.openqa.selenium.WebElement;
 
 import java.util.List;
 import java.util.Map;
+import javax.annotation.Nonnull;
 
 public class ContentMappedBy extends By {
     private final Map<ContentType, By> map;
+    private ContentType currentContent = NATIVE_MOBILE_SPECIFIC;
 
     public ContentMappedBy(Map<ContentType, By> map) {
         this.map = map;
     }
 
+    /**
+     * This method sets required content type for the further searching.
+     * @param type required content type {@link ContentType}
+     * @return self-reference.
+     */
+    public By useContent(@Nonnull ContentType type) {
+        checkNotNull(type);
+        currentContent = type;
+        return this;
+    }
+
     @Override public WebElement findElement(SearchContext context) {
-        return context.findElement(map.get(getCurrentContentType(context)));
+        return context.findElement(map.get(currentContent));
```

Comment:
is it ok if `map.get` returns null in any of these methods?

Reply:
It is not possible. It will use NATIVE_MOBILE_SPECIFIC as it is defined by default or it will use HTML_OR_DEFAULT when there is no `SearchContext` which also implements `ContextArare` or `HasSessionDetails`. Also it will use HTML_OR_DEFAULT  when `HasSessionDetails.isBrowser` returns `true` or current context != `NATIVE_CONTEXT`.

Also there is the method 
```java
    /**
     * This method sets required content type for the further searching.
     * @param type required content type {@link ContentType}
     * @return self-reference.
     */
    public By useContent(@Nonnull ContentType type) {
        checkNotNull(type);
        currentContent = type;
        return this;
    }
```